samcli/commands/local/cli_common/user_exceptions.py

```diff
@@ -66,3 +66,10 @@ class UnsupportedIntrinsic(UserException):
     Value from a template has an Intrinsic that is unsupported
     """
     pass
+
+
+class S3PermissionsRequired(UserException):
```

Comment:
Is this defined under the local command due to historical reasons?

Reply:
No these are exceptions for the local suite of commands. The S3PermissionRequired should not live here.